FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatStorage.swift

```diff
@@ -51,20 +51,43 @@ final class HeartbeatStorage: HeartbeatStorageProtocol {
   // MARK: - Instance Management
 
   /// Statically allocated cache of `HeartbeatStorage` instances keyed by string IDs.
-  private static var cachedInstances: [String: WeakContainer<HeartbeatStorage>] = [:]
+  #if compiler(>=6)
+    // In Swift 6, this property is not concurrency-safe because it is
+    // nonisolated global shared mutable state. Because this target's
+    // deployment version does not support Swift concurrency, it is marked as
+    // `nonisolated(unsafe)` to disable concurrency-safety checks. The
```

Comment:
Thanks for adding the comments! The name `nonisolated(unsafe)` isn't as intuitive to me as `@unchecked` is for `Sendable` so it was helpful to have the description.

Reply:
Agreed!